libs/math/include/math/TVecHelpers.h

```diff
@@ -497,6 +497,27 @@ class TVecFunctions {
         return v;
     }
 
+    friend inline constexpr VECTOR<T> MATH_PURE mix(VECTOR<T> x, VECTOR<T> y, T a) noexcept {
+        return x * (VECTOR<T>(1) - a) + y * a;
+    }
+
+    friend inline constexpr VECTOR<T> MATH_PURE lerp(VECTOR<T> x, VECTOR<T> y, T a) noexcept {
+        return mix(x, y, a);
+    }
+
+    // Note that the mix/lerp in <scalar.h> already covers the following two cases, but they are
```

Comment:
What happens if you include both `scalar.h` and `vec*.h`? Will it properly pick the more specialized variant from `vec*.h`?

Reply:
Yes, I confirmed this with a quick hack test.